akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/AccountExampleOneLinersWithNull.java

```diff
@@ -0,0 +1,159 @@
+        return Effect().persist(new Deposited(deposit.amount));
+    }
+
+    private Effect<AccountEvent, Account> withdrawCommand(OpenedAccount account, Withdraw withdraw) {
+        if ((account.balance - withdraw.amount) < 0.0) {
+            return Effect().unhandled(); // TODO replies are missing in this example
+        } else {
+            return Effect().persist(new Withdrawn(withdraw.amount))
+                    .andThen(acc2 -> { // FIXME in scaladsl it's named thenRun, change javadsl also?
+                        // we know this cast is safe, but somewhat ugly
+                        OpenedAccount openAccount = (OpenedAccount) acc2;
+                        // do some side-effect using balance
+                        System.out.println(openAccount.balance);
+                    });
+        }
+    }
+
+    private Effect<AccountEvent, Account> closeCommand(OpenedAccount account, CloseAccount cmd) {
+        if (account.balance == 0.0)
+            return Effect().persist(new AccountClosed());
+        else
+            return Effect().unhandled();
+    }
+
+
+    private CommandHandlerBuilder<AccountCommand, AccountEvent, Account, Account> initialHandler() {
+        return commandHandlerBuilder(Objects::isNull)
+                .matchCommand(CreateAccount.class, this::createAccount);
+    }
+
+    private CommandHandlerBuilder<AccountCommand, AccountEvent, OpenedAccount, Account> openedAccountHandler() {
+        return commandHandlerBuilder(OpenedAccount.class)
+                .matchCommand(Deposit.class, this::depositCommand)
+                .matchCommand(Withdraw.class, this::withdrawCommand)
+                .matchCommand(CloseAccount.class, this::closeCommand);
+    }
+
+    private CommandHandlerBuilder<AccountCommand, AccountEvent, ClosedAccount, Account> closedHandler() {
+        return commandHandlerBuilder(ClosedAccount.class)
+                .matchCommand(AccountCommand.class, () -> Effect().unhandled());
+    }
+
+    @Override
+    public CommandHandler<AccountCommand, AccountEvent, Account> commandHandler() {
+        return initialHandler()
+                .orElse(openedAccountHandler())
+                .orElse(closedHandler())
+                .build();
+    }
+
+
+    private OpenedAccount openAccount() {
+        return new OpenedAccount(0.0);
+    }
+
+    private OpenedAccount makeDeposit(OpenedAccount acc, Deposited deposit) {
+        return new OpenedAccount(acc.balance + deposit.amount);
+    }
+
+    private OpenedAccount makeWithdraw(OpenedAccount acc, Withdrawn withdrawn) {
+        return new OpenedAccount(acc.balance - withdrawn.amount);
+    }
+
+    private ClosedAccount closeAccount() {
+        return new ClosedAccount();
+    }
+
+    @Override
+    public EventHandler<Account, AccountEvent> eventHandler() {
+        return eventHandlerBuilder()
+                .matchEvent(AccountCreated.class, this::openAccount)
```

Comment:
This is wrong. 

When using `null`, we don't the means to add a match for the `null` state. 

This will make the `AccountCreated` event to be applied over any state.